subprojects/gdk_rust/gdk_common/src/be/transaction.rs

```diff
@@ -0,0 +1,716 @@
+                let new_in = elements::TxIn {
+                    previous_output: outpoint,
+                    is_pegin: false,
+                    has_issuance: false,
+                    script_sig: Script::default(),
+                    sequence: 0xffff_fffe, // nSequence is disabled, nLocktime is enabled, RBF is not signaled.
+                    asset_issuance: Default::default(),
+                    witness: TxInWitness::default(),
+                };
+                tx.input.push(new_in);
+            }
+            _ => panic!("unexpected mix of bitcoin and elements types"),
+        }
+    }
+
+    /// calculate transaction fee,
+    /// for bitcoin it requires all previous output to get input values.
+    /// for elements,
+    ///     for complete transactions looks at the explicit fee output,
+    ///     for incomplete tx (without explicit fee output) take the sum previous outputs value, previously unblinded
+    ///                       and use the outputs value that must be still unblinded
+    pub fn fee(
+        &self,
+        all_txs: &BETransactions,
+        all_unblinded: &HashMap<elements::OutPoint, Unblinded>,
+    ) -> u64 {
+        match self {
+            Self::Bitcoin(tx) => {
+                let sum_inputs = sum_inputs(tx, all_txs);
+                let sum_outputs: u64 = tx.output.iter().map(|o| o.value).sum();
+                sum_inputs - sum_outputs
+            }
+            Self::Elements(tx) => {
+                let has_fee = tx.output.iter().any(|o| o.is_fee());
+                if has_fee {
+                    tx.output
+                        .iter()
+                        .filter(|o| o.is_fee()) // TODO should check the asset is the policy asset
+                        .map(|o| o.minimum_value()) // minimum_value used for extracting the explicit value (value is always explicit for fee)
+                        .sum::<u64>()
+                } else {
+                    // while we are not filtering assets, the following holds for valid tx because
+                    // sum of input assets = sum of output assets
+                    let sum_outputs: u64 = tx.output.iter().map(|o| o.minimum_value()).sum();
+                    let sum_inputs: u64 = tx
+                        .input
+                        .iter()
+                        .map(|i| BEOutPoint::Elements(i.previous_output))
+                        .filter_map(|o| all_txs.get_previous_output_value(&o, all_unblinded))
+                        .sum();
+
+                    sum_inputs - sum_outputs
+                }
+            }
+        }
+    }
+
+    pub fn rbf_optin(&self) -> bool {
+        match self {
+            Self::Bitcoin(tx) => tx.input.iter().any(|e| e.sequence < 0xffff_fffe),
+            Self::Elements(tx) => tx.input.iter().any(|e| e.sequence < 0xffff_fffe),
+        }
+    }
+
+    pub fn is_redeposit(&self, all_scripts: &HashSet<Script>, all_txs: &BETransactions) -> bool {
```

Comment:
I think it's in the interface historically because green needed to refresh the locktime of outputs so this kind of tx was pretty common in a green wallet.
But it makes a broader sense. In the end, CPFP and consolidation are redepositing. So I think now the definition is the latter

Reply:
Sounds good, assuming the way this visually shows up in the GUI doesn't mention anything related to csv/2fa.

Reply:
in the GUI is represented as a "returning arrow" instead of "forward arrow" for outgoing or "backward arrow" for incoming, so no mention of csv/2fa